packages/commerce-sdk-react/src/hooks/useDNT.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useState} from 'react'
+import useAuthContext from './useAuthContext'
+import {getDefaultCookieAttributes} from '../utils'
+import Cookies from 'js-cookie'
+
+interface useDntReturn {
+    dntNotSet: boolean
+    updateDNT: (preference: boolean | null) => Promise<void>
+}
+
+/**
+ * Hook that updates the DNT preference and refreshes access token.
+ *
+ * @group Helpers
+ * @category Shopper Authentication
+ *
+ */
+const useDNT = (): useDntReturn => {
+    const auth = useAuthContext()
+
+    const dwDntValue = Cookies.get('dw_dnt')
```

Comment:
Based on Yuna's PR #1987, dw_dnt is set into cookie. We should read that value from auth